scalopus_tracing/src/native/tracepoint_collector_native.h

```diff
@@ -116,25 +117,17 @@ using NamedCounter = std::tuple<std::string, unsigned int>;
 
 /**
  * @brief A singleton class that keeps track of the ringbuffer allocated to each thread to insert tracepoints into.
+ *        If a thread goes out of scope the ringbuffer for that thread goes into the orphaned collection. This ensures
+ *        that no events get lost when a thread goes out of scope before its events are exfiltrated by the endpoint
+ *        native trace sender.
  */
-class TracePointCollectorNative : public MapTracker<unsigned long, tracepoint_collector_types::ScopeBufferPtr>
+class TracePointCollectorNative
```

Comment:
Shuffled public/private order, and instead of subclassing we now just allocate the `MapTracker` as a private class member.